lib/ansible/modules/database/postgresql/postgresql_tablespace.py

```diff
@@ -0,0 +1,557 @@
+
+        # For 9.1 version and earlier:
+        location = self.__exec_sql("SELECT 1 FROM information_schema.columns "
+                                   "WHERE table_name = 'pg_tablespace' "
+                                   "AND column_name = 'spclocation'", add_to_executed=False)
+        if location:
+            location = 'spclocation'
+        else:
+            location = 'pg_tablespace_location(t.oid)'
+
+        if not opt:
+            self.opt_not_supported = True
+            query = ("SELECT r.rolname, (SELECT Null), %s "
+                     "FROM pg_catalog.pg_tablespace AS t "
+                     "JOIN pg_catalog.pg_roles AS r "
+                     "ON t.spcowner = r.oid "
+                     "WHERE t.spcname = '%s'" % (location, self.name))
+        else:
+            query = ("SELECT r.rolname, t.spcoptions, %s "
+                     "FROM pg_catalog.pg_tablespace AS t "
+                     "JOIN pg_catalog.pg_roles AS r "
+                     "ON t.spcowner = r.oid "
+                     "WHERE t.spcname = '%s'" % (location, self.name))
+
+        res = self.__exec_sql(query, add_to_executed=False)
+
+        if not res:
+            self.exists = False
+            return False
+
+        if res[0][0]:
+            self.exists = True
+            self.owner = res[0][0]
+
+            if res[0][1]:
+                # Options exist:
+                for i in res[0][1]:
+                    i = i.split('=')
+                    self.settings[i[0]] = i[1]
+
+            if res[0][2]:
+                # Location exists:
+                self.location = res[0][2]
+
+    def create(self, location):
+        query = ("CREATE TABLESPACE %s LOCATION '%s'" % (pg_quote_identifier(self.name, 'database'), location))
+        return self.__exec_sql(query, ddl=True)
+
+    def drop(self):
+        return self.__exec_sql("DROP TABLESPACE %s" % pg_quote_identifier(self.name, 'database'), ddl=True)
+
+    def set_owner(self, new_owner):
+        if new_owner == self.owner:
+            return False
+
+        query = "ALTER TABLESPACE %s OWNER TO %s" % (pg_quote_identifier(self.name, 'database'), new_owner)
+        return self.__exec_sql(query, ddl=True)
+
+    def rename(self, newname):
+        query = "ALTER TABLESPACE %s RENAME TO %s" % (pg_quote_identifier(self.name, 'database'), newname)
+        self.new_name = newname
+        return self.__exec_sql(query, ddl=True)
+
+    def set_settings(self, new_settings):
+        # settings must be a dict {'key': 'value'}
+        if self.opt_not_supported:
+            return False
+
+        changed = False
+
+        # Apply new settings:
+        for i in new_settings:
+            if new_settings[i] == 'reset':
+                if i in self.settings:
+                    changed = self.__reset_setting(i)
+                    self.settings[i] = None
+
+            elif i not in self.settings:
```

Comment:
Can we rewrite this condition to something like this? `elif (i not in self.settings) or (str(new_settings[i]) != self.settings[i]):`